src/simsopt/geo/surfaceobjectives.py

```diff
@@ -180,10 +182,9 @@ def boozer_surface_residual(surface, iota, G, biotsavart, derivatives=0):
     dB_by_dX = biotsavart.dB_by_dX().reshape((nphi, ntheta, 3, 3))
     dB_dc = np.einsum('ijkl,ijkm->ijlm', dB_by_dX, dx_dc)
 
-    dresidual_dc = G*dB_dc \
-        - 2*np.sum(B[..., None]*dB_dc, axis=2)[:, :, None, :] * tang[..., None] \
-        - np.sum(B**2, axis=2)[..., None, None] * (dxphi_dc + iota * dxtheta_dc)
-    dresidual_diota = -np.sum(B**2, axis=2)[..., None] * xtheta
+    # dresidual_dc = G*dB_dc - 2*np.sum(B[..., None]*dB_dc, axis=2)[:, :, None, :] * tang[..., None] - B2[..., None, None] * (dxphi_dc + iota * dxtheta_dc)
+    dresidual_dc = sopp.boozer_dresidual_dc(G, dB_dc, B, tang, B2, dxphi_dc, iota, dxtheta_dc)
```

Comment:
I don't think this is particularly FLOP heavy, but instead mostly bandwidth bound, so I'm skeptical whether SIMD would do anything. I may be wrong about that, but even then it's probably not worth the effort for now, since in the very best case we'd only reduce runtime by like 15% (assuming we speed up that part of the code by a factor of 4 with SIMD).